modules/optimization/src/utils/ParameterMesh.C

```diff
@@ -14,22 +14,86 @@
 #include "libmesh/elem.h"
 #include "libmesh/fe_compute_data.h"
 #include "libmesh/fe_interface.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/explicit_system.h"
 
-ParameterMesh::ParameterMesh(const FEType & param_type, const std::string & exodus_mesh)
+ParameterMesh::ParameterMesh(const FEType & param_type,
+                             const std::string & exodus_mesh,
+                             const std::vector<std::string> & var_names)
   : _communicator(MPI_COMM_SELF), _mesh(_communicator)
 {
   _mesh.allow_renumbering(false);
-  _mesh.read(exodus_mesh);
-
+  _mesh.prepare_for_use();
+  _exodusII_io = std::make_unique<ExodusII_IO>(_mesh);
+  _exodusII_io->read(exodus_mesh);
+  _mesh.read(exodus_mesh); // fixme lynn,  I think I still need this
```

Comment:
document why

Reply:
I deleted the fixme comment.  I need this, without it the mesh doesn't get the data.